src/UnoFlipModel.java

```diff
@@ -299,10 +300,15 @@ public void playTurn(int btnIndex){        // add a
             chosenCardIndex = btnIndex;
 
             if (chosenCardIndex == -1) { // SELF DRAW ONE
-                turnSeqs.get(14).executeSequence(null);
-                status = " ";
-                notifyViews();
-                turnFinished = true;
+                if (validSelfDrawOne()){
+                    turnSeqs.get(14).executeSequence(null);
```

Comment:
magic number 14, executingSequence(null), don't really know what this means. needs commenting and a constant/some other way to reference for 14 if it is being used frequently